awx/api/serializers.py

```diff
@@ -4868,6 +4868,24 @@ def validate_node_state(self, value):
 
         return value
 
+    def validate_hostname(self, value):
+        if self.instance and self.instance.hostname != value:
+            raise serializers.ValidationError("Cannot change hostname.")
+
+        return value
+
+    def validate_listener_port(self, value):
+        if self.instance and self.instance.listener_port != value:
+            raise serializers.ValidationError("Cannot change listener port.")
+
+        return value
+
+
+class InstanceUpdateSerializer(InstanceSerializer):
```

Comment:
This shouldn't be necessary, IMO.  `hostname` and `listener_port` are covered by your `validate_*` methods above, and `node_type` was already handled previously.

Reply:
fixed